codalab/worker/worker.py

```diff
@@ -90,6 +90,7 @@ def __init__(
         self.idle_seconds = idle_seconds
 
         self.stop = False
+        self.terminating = False
```

Comment:
Document what the semantics of `terminating` and `stop` are, and what the flow is somewhere.